core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -102,7 +103,31 @@ public static WeekActivity createInstance(UserAnonymized userAnonymized, Goal go
 	public static WeekActivity createInstanceInactivity(UserAnonymized userAnonymized, Goal goal, ZonedDateTime startOfWeek)
 	{
 		WeekActivity result = createInstance(userAnonymized, goal, startOfWeek);
-		// leave days out
+		// add the days
+		// notice this doesn't take care of user time zone changes during the week
+		// so for consistency it is important that the batch script adding inactivity does so
+		for (int i = 0; i < 7; i++)
+		{
+			ZonedDateTime startOfDay = startOfWeek.plusDays(i);
+			if (isInFuture(startOfDay, startOfWeek.getZone()))
+			{
+				break;
+			}
+			if (wasGoalActiveAtDay(goal, startOfDay))
+			{
+				result.addDayActivity(DayActivity.createInstanceInactivity(userAnonymized, goal, startOfDay));
+			}
+		}
 		return result;
 	}
+
+	public static boolean wasGoalActiveAtDay(Goal goal, ZonedDateTime startOfDay)
```

Comment:
If you implement ``wasActive(Goal goal, ZonedDateTime startOfDay)`` on ``Goal``, then you can also reuse it from ``ActivityService``, see later comment.